tests/testthat/test_merModList.R

```diff
@@ -1,113 +1,126 @@
-                                      replace=FALSE),]
-  }
-
-  g1 <- glmerModList(formula = form,
-                    data = modDat, family = "binomial",
-                    control = glmerControl(optimizer="bobyqa",
-                                           optCtrl=list(maxfun = 1e6)))
-  g1T <- glmer(form, family = "binomial", data = grouseticks,
-               control = glmerControl(optimizer="bobyqa",
-                                      optCtrl=list(maxfun = 1e6)))
-
-  expect_equal(VarCorr(g1)$stddev$BROOD, attr(VarCorr(g1T)$BROOD, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$stddev$YEAR, attr(VarCorr(g1T)$YEAR, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$BROOD, attr(VarCorr(g1T)$BROOD, "corre"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$YEAR, attr(VarCorr(g1T)$YEAR, "corre"),
-               tolerance = 0.0001)
-
-  form <- TICKS_BIN ~ HEIGHT +(1|BROOD)
-  g1 <- glmerModList(formula = form,
-                     data = modDat, family = "binomial",
-                     control = glmerControl(optimizer="bobyqa",
-                                            optCtrl=list(maxfun = 1e6)))
-  g1T <- glmer(form, family = "binomial", data = grouseticks,
-               control = glmerControl(optimizer="bobyqa",
-                                      optCtrl=list(maxfun = 1e6)))
-
-  expect_equal(VarCorr(g1)$stddev$BROOD, attr(VarCorr(g1T)$BROOD, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$stddev$YEAR, attr(VarCorr(g1T)$YEAR, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$BROOD, attr(VarCorr(g1T)$BROOD, "corre"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$YEAR, attr(VarCorr(g1T)$YEAR, "corre"),
-               tolerance = 0.0001)
-
-})
-
-#ICC function----
-context("ICC function")
-
-test_that("ICC function works", {
-  ICC1 <- ICC(outcome = "Reaction", group = "Subject", data = sleepstudy)
-  expect_is(ICC1, "numeric")
-  expect_equal(ICC1, 0.3948896, tol = .001)
-})
+# test merModList functions
+
+#Do merModList objects get built and work----
+context("Do merModList objects get built and work")
+
+old_warn <- getOption("warn")
+options(warn= -1)
+
+test_that("simple cases work", {
+  # skip_on_cran()
+  library(blme)
+  d <- expand.grid(fac1=LETTERS[1:5], grp=factor(1:10),
+                   obs=1:100)
+  split <- sample(x = LETTERS[9:15], size = nrow(d), replace=TRUE)
+  d$y <- simulate(~fac1+(1|grp),family = gaussian,
+                  newdata=d,
+                  newparams=list(beta=c(2,1,3,4,7), theta=c(.25),
+                                 sigma = c(.23)))[[1]]
+  out <- split(d, split)
+  rm(split)
+  g1 <- lmerModList(formula = y~fac1+(1|grp), data=out, 
+                    control= lmerControl(check.conv.grad = .makeCC("warning", tol= 8e-3)))
```

Comment:
Is this relaxing the tolerance or making it more strict compared to before? Or did the default change and this is making it work the old way?

Reply:
Relaxing the tolerance to avoid convergence warning messages during `devtools::test`. The test scripts are not dependent on precise convergence, but these emitted warning messages give false positives on code problems